File: 2020_11_09_arrays_pop_front.py
# PopFront
# Given an array, remove and return the value at
# the beginning of the array. Do this without using
# any built-in array methods except pop() .
def popFront(myList):
    # copy all the array value by one to the left
    # then, pop the last one
    l = len(myList)
    temp = myList[0]
    for i in range(1, l, 1):
        myList[i-1] = myList[i]
    myList.pop()
    print(myList)
    return temp

def popFront2(myList):
    temp = myList[0]
    myList.pop(0)
    print(myList)
    return temp

# print(popFront([5,3,4,2,8]))
print(popFront2([5,3,4,2,8]))

File: 2020_11_09_arrays_pop_front.js
// PopFront
// Given an array, remove and return the value at
// the beginning of the array. Do this without using
// any built-in array methods except pop() .
function popFront(arr) {
    //copy all the array value by one to the left
    //then, pop the last one
    let temp = arr[0];
    let len = arr.length;
    for (let i = 0; i < len-1; i++) {
        arr[i] = arr[i+1];
    }
    arr.pop();
    console.log(arr);
    return temp;
}
console.log(popFront([5,3,4,2,8]));

//built-in method:
let myArr = [5,3,4,2,8];
myArr.shift();
console.log(myArr);